version: "3.7"

x-app: &app
  ports:
    - "8000:8000"
  volumes:
    - ./src:/src
    - ./tests:/tests
    - ./deploy:/deploy
    - ./pyproject.toml:/pyproject.toml
  environment: &base-app-env
    SECRET_KEY: "insecure"
    DEBUG: "true"
    OIDC_BASE_URL: "https://login.microsoftonline.com/72fca1b1-2c2e-4376-a445-294d80196804"
    OIDC_RP_CLIENT_ID:
    OIDC_RP_CLIENT_SECRET:
    APPLICATIONINSIGHTS_CONNECTION_STRING:
  depends_on:
    - database


services:
  database:
    ports:
      - "5432"
    environment:
      POSTGRES_DB: bereikbaarheid
      POSTGRES_USER: bereikbaarheid
      POSTGRES_PASSWORD: insecure
    image: postgis/postgis:15-3.4
    networks:
      - backend

  app:
    <<: *app
    build:
      context: .
      target: app
    image: ${REGISTRY:-127.0.0.1:5001}/${REPOSITORY:-opdrachten/bereikbaarheid-backend-django}:${VERSION:-latest}
    command: /deploy/docker-run.sh

  dev:
    <<: *app
    build:
      context: .
      target: dev
    environment:
      <<: *base-app-env
    entrypoint: /deploy/wait-for-it.sh database:5432 --
    command:
      python manage.py runserver 0.0.0.0:8000
    networks:
      - backend

  test:
    <<: *app
    build:
      context: .
      target: tests
    environment:
      <<: *base-app-env
      DJANGO_SETTINGS_MODULE: "main.settings"
      OIDC_RP_CLIENT_ID: tests
      OIDC_RP_CLIENT_SECRET: tests
    entrypoint: /deploy/wait-for-it.sh database:5432 --
    command:
      pytest
    networks:
      - backend

networks:
  backend:
    name: amsterdam-bereikbaarheid

